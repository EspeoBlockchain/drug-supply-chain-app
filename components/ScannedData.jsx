import React from 'react';
import {
  View,
  Text,
  StyleSheet,
  ScrollView,
} from 'react-native';
import PropTypes from 'prop-types';
import colors from '../constants/Colors';
import fonts from '../constants/Fonts';

const styles = StyleSheet.create({
  data: {
    flex: 1,
    alignSelf: 'stretch',
  },
  container: {
    flex: 1,
    backgroundColor: colors.registerLight,
    padding: 20,
  },
  title: {
    color: colors.white,
    fontFamily: fonts.main,
    fontSize: 16,
    marginBottom: 5,
  },
  table: {
    borderColor: colors.white,
    borderWidth: 2,
    borderBottomWidth: 0,
    marginBottom: 50,
  },
  row: {
    flexDirection: 'row',
    alignItems: 'center',
    borderBottomWidth: 2,
    borderColor: colors.white,
  },
  label: {
    width: 150,
    padding: 5,
    color: colors.white,
    fontFamily: fonts.main,
  },
  value: {
    flex: 1,
    flexWrap: 'wrap',
    padding: 5,
    color: colors.white,
    borderLeftWidth: 2,
    borderColor: colors.white,
    fontFamily: fonts.main,
  },
});

const scannedData = (props) => {
  const { data } = props;

  const filteredData = Object.keys(data).map((item) => {
    let row = null;
    if (data[item] !== '') {
      row = (
        <View key={item} style={styles.row}>
          <Text style={styles.label}>
            {`${item}:`}
          </Text>
          <Text style={styles.value}>
            {data[item]}
          </Text>
        </View>
      );
    }
    return row;
  });

  return (
    <View style={styles.data}>
      <ScrollView style={styles.container}>
        <Text style={styles.title}>Your scanned data</Text>
        <View style={styles.table}>
          {filteredData}
        </View>
      </ScrollView>
    </View>
  );
};

scannedData.propTypes = {
  data: PropTypes.shape({
    drugItemId: PropTypes.string,
    address: PropTypes.string,
    participantCategory: PropTypes.oneOfType([
      PropTypes.number,
      PropTypes.string,
    ]),
    temperature: PropTypes.oneOfType([
      PropTypes.number,
      PropTypes.string,
    ]),
    transitCategory: PropTypes.oneOfType([
      PropTypes.number,
      PropTypes.string,
    ]),
    privateKey: PropTypes.string,
    publicKey: PropTypes.string,
  }),
};

scannedData.defaultProps = {
  data: {},
};

export default scannedData;
