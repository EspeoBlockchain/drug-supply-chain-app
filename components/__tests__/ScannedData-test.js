import 'react-native';
import React from 'react';
import ScannedData from '../ScannedData';
import renderer from 'react-test-renderer';

it('renders corectly', () => {
    const tree = renderer.create(<ScannedData />).toJSON();

    expect(tree).toMatchSnapshot();
});