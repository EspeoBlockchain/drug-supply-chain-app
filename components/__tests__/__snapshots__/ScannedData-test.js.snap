// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders corectly 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "flex": 1,
    }
  }
/>
`;
